forest/bonsai/simulate_gps_data.py

```diff
@@ -1117,31 +1123,306 @@ def process_switches(
+    Returns:
+        addresses: (np.ndarray) contains address coordinates of each person
+    Raises:
+        RuntimeError: if the api raises error for too many tries
+        ValueError: if the api returns no results
+    """
+
+    overpy_query = f"""
+    [out:json];
+    area["ISO3166-1"="{country}"][admin_level=2] -> .country;
+    area["name"="{city}"] -> .city;
+    node(area.country)(area.city)["addr:street"];
+    out center 150;
+    """
+
+    for try_no in range(4):
+        response = requests.get(
+            OVERPASS_URL,
+            params={"data": overpy_query}, timeout=5 * 60
+        )
+        if try_no == 3:
+            raise RuntimeError(
+                "Too many Overpass requests in a short time. Please try again in a minute."
+                )
+        elif response.status_code == 200:
+            break
+        else:
+            time.sleep(30)
+
+    res = response.json()
+    try:
+        index = np.random.choice(
+            range(len(res["elements"])), 100, replace=False
+        )
+    except ValueError:
+        raise ValueError(
+            "Overpass query came back empty. Check the location argument, ISO code and city name, for any misspellings."
+            )
+
+    nodes = np.array(res["elements"])[index]
+
+    return nodes
+
+
+def generate_nodes(
+    house_address: Tuple[float, float],
+    employment: Occupation
+) -> Dict[str, List[Tuple[float, float]]]:
+    """Generates multiple amenities coordinates.
+
+    Args:
+        house_address: (tuple) coordinates of the house
+        employment: (Occupation) occupation of the person
+    Returns:
+        nodes: (dictionary) contains coordinates of each amenity
+    Raises:
+        RuntimeError: if the api raises error for too many tries
+    """
+
+    house_area = bounding_box(house_address, 2000)
+    house_area2 = bounding_box(house_address, 3000)
+
+    q_employment = ""
+    if employment == Occupation.WORK:
+        q_employment = f'node{house_area}["office"];'
+    elif employment == Occupation.SCHOOL:
+        q_employment = f'node{house_area2}["amenity"="university"];\n\t\t\tway{house_area2}["amenity"="university"]'
+
+    overpy_query2 = f"""
+    [out:json];
+    (
+        node{house_area}["amenity"="cafe"];
+        node{house_area}["amenity"="bar"];
+        node{house_area}["amenity"="restaurant"];
+        node{house_area}["amenity"="cinema"];
+        node{house_area}["leisure"="park"];
+        node{house_area}["leisure"="dance"];
+        node{house_area}["leisure"="fitness_centre"];
+        way{house_area}["amenity"="cafe"];
+        way{house_area}["amenity"="bar"];
+        way{house_area}["amenity"="restaurant"];
+        way{house_area}["amenity"="cinema"];
+        way{house_area}["leisure"="park"];
+        way{house_area}["leisure"="dance"];
+        way{house_area}["leisure"="fitness_centre"];
+        {q_employment}
+    );
+    out center;
+    """
+
+    for try_no in range(4):
```

Comment:
I'd let the user of this code handle the retry logic and may be add a comment to the function docstring with a retry strategy suggestions.